AirportTicketBookingSystem/AirportTicketBookingSystem/Services/BookingService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using AirportTicketBookingSystem.Enums;
+using AirportTicketBookingSystem.Interfaces;
+using AirportTicketBookingSystem.Models;
+
+namespace AirportTicketBookingSystem.Services
+{
+    public class BookingService : BaseService<Booking>, IBookingService
+    {
+        private readonly IPassengerRepository _passengerRepository;
+        private readonly IBookingRepository _bookingRepository;
+
+        public BookingService(IBookingRepository bookingRepo, IPassengerRepository passengerRepo) : base(bookingRepo)
+        {
+            _passengerRepository = passengerRepo;
+            _bookingRepository = bookingRepo;
+        }
+
+        public List<Booking> GetBookingsForPassenger(int passengerId)
+        {
+            var passenger = _passengerRepository.GetById(passengerId);
+            if (passenger == null)
+                throw new KeyNotFoundException($"Passenger with Id {passengerId} does not exist");
+
+            var allBookings = _bookingRepository.GetAll();
+            return allBookings.Where(b => b.PassengerId == passengerId).ToList();
```

Comment:
filtration should happen on the db level, you shouldn't get all bookings here and do then the filtration. Add a new repo method to get booking by passenger id